src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -1104,10 +1238,15 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
         }
 
         protected virtual TagBuilder GenerateLink(
-            [NotNull] string linkText,
-            [NotNull] string url,
+            string linkText,
+            string url,
```

Comment:
I'm not sure about that. Isn't this supposed to throw?

`IUrlHelper` can indeed return null.